contrib/babelfishpg_tsql/src/pltsql_identity.c

```
@@ -470,3 +472,100 @@ pltsql_revert_last_scope_identity(int nest_level)
 	last_used_scope_seq_identity = old_top->prev;
 	pfree(old_top);
 }
+
+void
+reseed_identity_post_select_into(Oid relid)
+{
+	Oid            relowner;
+	Oid            save_userid;
+	int            save_sec_context;
+	int            saved_sql_dialect;
+	bool           is_identity_increasing = false;
+	const char     *schema_name = NULL;
+	const char     *table_name = NULL;
+	const char     *seq_name = NULL;
+	const char     *identity_colname = NULL;
+	Relation       rel;
+	TupleDesc      tupdesc;
+	StringInfoData querybuf;
+
+	rel = RelationIdGetRelation(relid);
+
+	relowner = rel->rd_rel->relowner;
+	schema_name = quote_identifier(get_namespace_name_or_temp(RelationGetNamespace(rel)));
+	table_name = quote_identifier(RelationGetRelationName(rel));
+
+	tupdesc = RelationGetDescr(rel);
+
+	for (AttrNumber attnum = 1; attnum <= RelationGetNumberOfAttributes(rel); attnum++)
+	{
+		Form_pg_attribute attr = TupleDescAttr(tupdesc, attnum - 1);
+		Oid               seq_relid;
+		HeapTuple         pgstuple;
+
+		if (!attr->attidentity)
+			continue;
+
+		seq_relid = getIdentitySequence(rel, attnum, false);
+
+		pgstuple = SearchSysCache1(SEQRELID, ObjectIdGetDatum(seq_relid));
+		if (!HeapTupleIsValid(pgstuple))
+			elog(ERROR, "cache lookup failed for sequence %u", relid);
+
+		is_identity_increasing = ((Form_pg_sequence) GETSTRUCT(pgstuple))->seqincrement > 0;
+		seq_name = quote_identifier(get_rel_name(seq_relid));
+		identity_colname = quote_identifier(NameStr(attr->attname));
+
+		ReleaseSysCache(pgstuple);
+		break;
+	}
+
+	RelationClose(rel);
+
+	if (!identity_colname || !OidIsValid(relowner))
```

in which can this could be true?

---

When new table does not have any identity.

---

and checking for `!OidIsValid(relowner)` just for safety.

---

if we are checking !OidIsValid(relowner) just for safety then it should throw an error instead of just returning silently

---

Done